qiskit_nature/second_q/operators/symmetric_two_body.py

```diff
@@ -0,0 +1,751 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022, 2023.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+# pylint: disable=invalid-name
+
+"""
+Symmetric 2-body electronic integrals (:mod:`qiskit_nature.second_q.operators.symmetric_two_body`)
+==================================================================================================
+
+.. currentmodule:: qiskit_nature.second_q.operators.symmetric_two_body
+
+This module provides utilities to deal with symmetry-reduced 2-body electronic integrals.
+
+Container classes
+-----------------
+
+.. autosummary::
+   :toctree: ../stubs/
+   :template: autosummary/class_with_inherited_members.rst
+   :nosignatures:
+
+   SymmetricTwoBodyIntegrals
+   S1Integrals
+   S4Integrals
+   S8Integrals
+
+Expansion methods
+-----------------
+
+These methods can be used to expand from lower to higher symmetries.
```

Comment:
I would have considered 8 fold symmetry higher than 4 fold - in that it has more symmetries. To me these are reducing the symmetries which expands the memory. Except for the first one which is taking a full matrix and creating an S1 from it.
I am not going to comment on the reduce since thats the opposite here, except to say the expand does not seem to be an inverse of reduce its more an any to S1 well not quite any just S4 and S8 (full to S1 is covered by expand.) I guess on reduce side we don't need the opposite of expand since any of these are perceived as a full array by and end user - and maybe some .array property does just that?

Reply:
Yes I see what you mean. I liked `expand` since this clearly states the expansion from lower to higher memory requirements. I thought `reduce` would be a good mnemonic to the opposite operation.

But you are right that "lower symmetry" correlates with "higher memory consumption".

And yes, it is not exactly the inverse. Expand really always goes up to `S1` whereas reduce tries its best. There are standalone methods for each conversion which a user can use to their liking if they want to achieve a specific symmetry.

So to summarize:
- `expand`: any to S1
- `reduce`: any to as low memory consumption as possible (highest symmetry) [I think you mixed this up above]

Do you think I can rephrase the docstring to talk about memory consumption or would this be too confusing?
Or do you have some better naming suggestions?

Reply:
I pushed a suggested rewording just now. Let me know what you think :+1: